torch/csrc/autograd/profiler_utils.cpp

```diff
@@ -115,23 +115,51 @@ std::unordered_map<std::string, c10::IValue> saveExtraArgs(const at::RecordFunct
 uint64_t computeFlops(const std::string &op_name, const std::unordered_map<std::string, c10::IValue> &extra_args) {
   if (op_name == kConv2dOp) {
     if (extra_args.find(kInputSize) == extra_args.end()
-        || extra_args.find(kWeightSize) == extra_args.end()) {
-      TORCH_WARN("Calculating flops for aten::conv2d requires input_size and weight_size in saved arguments.");
+        || extra_args.find(kWeightSize) == extra_args.end()
+        || extra_args.find(kGroups) == extra_args.end()
+        || extra_args.find(kPadding) == extra_args.end()
+        || extra_args.find(kStride) == extra_args.end()
+        || extra_args.find(kDilation) == extra_args.end()) {
+      TORCH_WARN("Calculating flops for aten::conv2d requires groups, padding, stride, dilation, input_size, and weight_size in saved arguments.");
       return 0;
     }
     auto input_sizes_ref = extra_args.at(kInputSize);
     auto kernel_sizes_ref = extra_args.at(kWeightSize);
+    auto groups_ref = extra_args.at(kGroups);
+    auto padding_ref = extra_args.at(kPadding);
+    auto stride_ref = extra_args.at(kStride);
+    auto dilation_ref = extra_args.at(kDilation);
     if (!input_sizes_ref.isIntList() || !kernel_sizes_ref.isIntList()) {
       TORCH_WARN("Failed to compute flops for op aten::conv2d because it requires input and weight tensor sizes.");
       return 0;
     }
+    if (!padding_ref.isIntList() || !stride_ref.isIntList() || !dilation_ref.isIntList()) {
+      TORCH_WARN("Failed to compute flops for op aten::conv2d because it requires padding, stride, and dilation values.");
+      return 0;
+    }
 
     const std::vector<int64_t> input_sizes = input_sizes_ref.toIntVector();
     const std::vector<int64_t> kernel_sizes = kernel_sizes_ref.toIntVector();
+    const uint64_t groups = groups_ref.toInt();
+    const std::vector<int64_t> padding = padding_ref.toIntVector();
+    const std::vector<int64_t> stride = stride_ref.toIntVector();
+    const std::vector<int64_t> dilation = dilation_ref.toIntVector();
     if (input_sizes.size() != 4 || kernel_sizes.size() != 4) {
       TORCH_WARN("Failed to compute flops for op aten::conv2d because both input and weight must be size 4.");
       return 0;
     }
+    if (!groups) {
+      TORCH_WARN("Failed to compute flops for op aten::conv2d because group size must not be 0.");
+      return 0;
+    }
+    if (padding.size() != 2 || dilation.size() != 2) {
```

Comment:
PyTorch doesn't support having a different padding for left and right (or top and bottom)? Sorry I'm more familiar with Caffe2 conv operator.

Reply:
According to torch.nn.conv2d doc, padding only supports number of points in two directions and is applied on both sides: https://pytorch.org/docs/stable/generated/torch.nn.Conv2d.html